sdk/storage/test/datalake/file_system_client_test.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "file_system_client_test.hpp"
+
+#include <algorithm>
+
+namespace Azure { namespace Storage { namespace Test {
+
+  const size_t c_PATH_TEST_SIZE = 5;
+
+  std::shared_ptr<DataLake::FileSystemClient> FileSystemClientTest::m_fileSystemClient;
+  std::string FileSystemClientTest::m_fileSystemName;
+
+  std::vector<std::string> FileSystemClientTest::m_PathNameSetA;
+  std::vector<std::string> FileSystemClientTest::m_PathNameSetB;
+  std::string FileSystemClientTest::m_DirectoryA;
+  std::string FileSystemClientTest::m_DirectoryB;
+
+  void FileSystemClientTest::SetUpTestSuite()
+  {
+    m_fileSystemName = LowercaseRandomString(10);
+    m_fileSystemClient = std::make_shared<DataLake::FileSystemClient>(
+        DataLake::FileSystemClient::CreateFromConnectionString(
+            ADLSGen2ConnectionString(), m_fileSystemName));
+    m_fileSystemClient->Create();
+
+    m_DirectoryA = LowercaseRandomString(10);
+    m_DirectoryB = LowercaseRandomString(10);
+    for (unsigned i = 0; i < c_PATH_TEST_SIZE; ++i)
+    {
+      {
+        auto name = m_DirectoryA + LowercaseRandomString(10);
+        m_fileSystemClient->GetPathClient(m_DirectoryA + "/" + name).CreateAsFile();
+        m_PathNameSetA.emplace_back(std::move(name));
+      }
+      {
+        auto name = m_DirectoryB + LowercaseRandomString(10);
+        m_fileSystemClient->GetPathClient(m_DirectoryB + "/" + name).CreateAsFile();
+        m_PathNameSetB.emplace_back(std::move(name));
+      }
+    }
+  }
+
+  void FileSystemClientTest::TearDownTestSuite() { m_fileSystemClient->Delete(); }
+
+  std::vector<DataLake::Path> FileSystemClientTest::ListAllPaths(
+      bool recursive,
+      const std::string& directory)
+  {
+    std::vector<DataLake::Path> result;
+    std::string continuation;
+    DataLake::ListPathsOptions options;
+    if (!directory.empty())
+    {
+      options.Directory = directory;
+    }
+    do
+    {
+      auto response = m_fileSystemClient->ListPaths(recursive, options);
+      result.insert(result.end(), response.Paths.begin(), response.Paths.end());
+      if (response.Continuation.HasValue())
+      {
+        continuation = response.Continuation.GetValue();
+        options.Continuation = continuation;
+      }
+    } while (!continuation.empty());
+    return result;
+  }
+
+  DataLake::DataLakeHttpHeaders FileSystemClientTest::GetInterestingHttpHeaders()
+  {
+    static DataLake::DataLakeHttpHeaders result{
+        std::string("no-cache"),
```

Comment:
aggregate initialization is forbidden.

Reply:
Well, I do. I cannot figure out which variable you're trying to assign the value to.

Reply:
Then can you please provide an easy solution for this code that inline initializing the result. This way for each call a copy is made and we don't need to re-initialize the strings.

Reply:
make it a member variable?

Reply:
Not ideal since the only place to initialize it is `SetUpTestSuite()` and it is called for each individual test.